src/lazy/binary/raw/v1_1/reader.rs

```diff
@@ -0,0 +1,141 @@
+#![allow(non_camel_case_types)]
+
+use crate::lazy::binary::raw::v1_1::immutable_buffer::ImmutableBuffer;
+use crate::lazy::decoder::{LazyDecoder, LazyRawReader};
+use crate::lazy::encoder::private::Sealed;
+use crate::lazy::encoding::BinaryEncoding_1_1;
+use crate::lazy::raw_stream_item::{LazyRawStreamItem, RawStreamItem};
+use crate::result::IonFailure;
+use crate::IonResult;
+
+use bumpalo::Bump as BumpAllocator;
+
+pub struct LazyRawBinaryReader_1_1<'data> {
+    data: ImmutableBuffer<'data>,
+}
+
+impl<'data> LazyRawBinaryReader_1_1<'data> {
+    fn new(data: &'data [u8]) -> Self {
+        Self::new_with_offset(data, 0)
+    }
+
+    fn new_with_offset(data: &'data [u8], offset: usize) -> Self {
+        let data = ImmutableBuffer::new_with_offset(data, offset);
+        Self { data }
+    }
+
+    fn read_ivm<'top>(
+        &mut self,
+        buffer: ImmutableBuffer<'data>,
+    ) -> IonResult<LazyRawStreamItem<'top, BinaryEncoding_1_1>>
+    where
+        'data: 'top,
+    {
+        let ((major, minor), _buffer_after_ivm) = buffer.read_ivm()?;
+        if (major, minor) != (1, 1) {
+            return IonResult::decoding_error(format!(
+                "unsupported version of Ion: v{}.{}; only 1.1 is supported by this reader",
```

Comment:
Doesn't need to be fixed now, but you should probably consider whether this design will make it possible to read any IVM. There needs to be some way to (mid stream) read either IVM and then switch to the correct reader.